src/variant_iter.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+// This is similar to the GVariantIter provided by glib, but that would
+// introduce a heap allocation and doesn't provide a way to determine how
+// many items are left in the iterator.
+
+use std::iter::{DoubleEndedIterator, ExactSizeIterator, Iterator};
+
+use variant::Variant;
+
+/// Iterator over items in a variant.
+#[derive(Debug)]
+pub struct VariantIter {
+    variant: Variant,
```

Comment:
Why not `&'a Variant`? We don't need to clone above, especially not because variants are immutable :)

Reply:
The clone just increases the reference count, right? This matches the behavior of `GVariantIter`, which calls `g_variant_ref`.

If `VariantIter` just takes a reference to the `Variant`, that should work the same way for most purposes. But it can sometimes be a pain to have the lifetime of one struct tied to that of another. I think avoiding that is probably worth the cost of just incrementing a reference count, while also being more consistent with the fact that most gtk-rs types (like gtk itself) wrap reference counting.

Reply:
Ok, works for me